app/script/event/EventRepository.coffee

```diff
@@ -355,7 +357,8 @@ class z.event.EventRepository
         return @cryptography_repository.decrypt_event event
         .catch (decrypt_error) =>
           # Get error information
-          error_code = decrypt_error.code or 999
+          decrypt_error.code = UNKNOWN_DECRYPT_ERROR_CODE if not decrypt_error.code
```

Comment:
Yes, Lipis is right. The function `_report_decrypt_error` uses the `decrypt_error.code` at:

```coffee
    custom_data =
      cryptobox_version: cryptobox.version
      client_local_class: @current_client().class
      client_local_type: @current_client().type
      error_code: decrypt_error.code
      event_type: event.type
      session_id: session_id
```

So I thought it might be good adding the `999` as `code` property to the `decrypt_error` object.

Why do you think that it's masking unknown errors? The `999` is only added in case it's an error without a code which means that it is unknown to us.